lib/oli/delivery/paywall.ex

```diff
@@ -404,6 +404,24 @@ defmodule Oli.Delivery.Paywall do
     end
   end
 
+  @spec has_payment_codes?(integer()) :: boolean()
+  @doc """
+  Return if a product has any payment codes.
+  """
+  def has_payment_codes?(product_id) do
+    count =
+      Repo.one(
+        from(p in Payment,
```

Comment:
There is likely a more DB efficient way of determining if at least 1 payment code exists.  Here, the query counts all records, which requires a pass over all of the matching records.  More efficient could be to query `Repo.all`, without the aggregation of the Count, and adding simply a `limit: 1`